packages/scanner-global-library/src/page-network-tracer.ts

```diff
@@ -0,0 +1,155 @@
+                    url: request.url(),
+                });
+
+                this.logger.logInfo(`[Network] Processing URL`, { traceUrl: request.url() });
+
+                if (!request.isInterceptResolutionHandled()) {
+                    await request.continue();
+                }
+            } catch (error) {
+                this.logger.logError(`The 'request' page event handle failed`, {
+                    traceUrl: request.url(),
+                    error: System.serializeError(error),
+                });
+            }
+        };
+        this.setPageEventHandler('request', pageOnRequestEventHandler, page);
+
+        const pageOnRequestFinishedEventHandler = async (request: Puppeteer.HTTPRequest) => {
+            try {
+                const response = request.response();
+                const serverResponseTiming = response.timing()?.receiveHeadersEnd - response.timing()?.sendStart;
+                const data = {
+                    status: 'completed',
+                    url: request.url(),
+                    httpStatus: `${response.status()} ${response.statusText()}`,
+                    requestHeaders: request.headers(),
+                    responseHeaders: response.headers(),
+                    serverResponseTiming,
+                };
+                const traceRequest = this.networkTraceData.requests.find((r) => r.url === request.url());
+                if (traceRequest) {
+                    traceRequest.completed = true;
+                }
+
+                this.logger.logInfo(`[Network] Request completed`, { data: JSON.stringify(data, undefined, 2) });
+                if (!request.isInterceptResolutionHandled()) {
+                    await request.continue();
+                }
+            } catch (error) {
+                this.logger.logError(`The 'requestfinished' page event handle failed`, {
+                    traceUrl: request.url(),
+                    error: System.serializeError(error),
+                });
+            }
+        };
+        this.setPageEventHandler('requestfinished', pageOnRequestFinishedEventHandler, page);
+
+        const pageOnRequestFailedEventHandler = async (request: Puppeteer.HTTPRequest) => {
+            try {
+                const response = request.response();
+                const serverResponseTiming = response.timing()?.receiveHeadersEnd - response.timing()?.sendStart;
+                const data = {
+                    status: 'failed',
+                    url: request.url(),
+                    httpStatus: `${response?.status()} ${response?.statusText()}`,
+                    requestHeaders: request.headers(),
+                    responseHeaders: response?.headers(),
+                    serverResponseTiming,
+                };
+                const traceRequest = this.networkTraceData.requests.find((r) => r.url === request.url());
+                if (traceRequest) {
+                    traceRequest.completed = true;
```

Comment:
no, try/catch is to bubble up error from event execution scope to main process, expected handler to run w/o error. We need trace data rather than completed flag to be set.

Reply:
Means if something threw we want the trace to remain pending so we will see it?

Reply:
yes, we want to see error, pending will be ok